### DISCUSSION TITLE
How to extract the cst from the LIVER mat file and use it for the external dicom mask

### DISCUSSION THREAD

**Discussion by Yuri-HubL** [e0404/matRad#670] [Q&A]
Dear Administrator

I used the matRad-dev_varRBErobOpt branch to generate an executable TOPAS file. After running TOPAS, I got a dcm file.
![YValue dicom](https://github.com/e0404/matRad/assets/72288469/50f082f3-4d40-44c6-8165-bcd65e7dc64f)

Now I want to remove all values outside of matRad's skin. How do I do that?
From the image below we can see that there are a lot of values outside the body. I want to use the skin as a mask to remove all the data outside the mask file in the dicom file.
![mask](https://github.com/e0404/matRad/assets/72288469/07f22367-e9b3-42ec-bf82-ec83690065f0)

However, I ran into a problem, because I used the cst{15, 4}{1, 1} index as the mask, and the dimensions were not the same as the dicom one, so I couldn't do the multiplication to subtract out the values other than YValue.

I will attach my dicom file and the execution code.
[matRad-mat-dicom-match.txt](https://github.com/e0404/matRad/files/13246442/matRad-mat-dicom-match.txt)
DICOM file: https://drive.google.com/file/d/10zGZK5qRsxBikgUiyQSccjn_zF8V83m0/view?usp=sharing
Looking forward to your reply

your sincerely,
Yuri

**Comment by remocristoforetti** [comment]
Hi Yuri,
thank you for your question. The `Topas` interface automatically matches the scorer resolution in `Topas` with the specified dose grid resolution set in `matRad`. The indexes stored into the `cst` structure refer instead to the CT resolution.

Probably, the reason why you see the resolution mismatch is because the CT resolution and the dose grid resolution are different.
In order to solve the issue, one possible approach could be that of resizing the `cst` structure to the desired dose grid resolution (take a look at the function matRad_resizeCstToGrid).

Hope this can help,
Best regards,
Remo
